GoogleSignInSwift/Sources/GIDSwiftUISignInButton.swift

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import CoreGraphics
+
+// MARK: - Sign-In Button
+
+/// A Google Sign In button to be used in SwiftUI.
+@available(iOS 13.0, macOS 10.15, *)
+public struct GIDSwiftUISignInButton: View {
```

Comment:
It looks like ObjC interop isn't possible for structs.  Are we OK with that?  Would it be possible to use a class here?

Reply:
Do you have a use-case for this? We have an existing button working in ObjC/UIKit right now, and so using this SwiftUI button over there shouldn't be a need. Moreover, I believe SwiftUI enforces conformers of `View` to be value types. Using a `final class` (e.g., `final class MyView: View {...}`) will yield an error: "Fatal error: views must be value types (either a struct or an enum)."

If we want to expose this button to UIKit, then we'd need to use [`UIHostingController`](https://developer.apple.com/documentation/swiftui/uihostingcontroller) and [`NSHostingView`](https://developer.apple.com/documentation/swiftui/nshostingview).

Reply:
I'm thinking of the AppKit use case.  A traditional ObjC macOS app that wants to use the SwiftUI button.  It's an edge case, but it'd be nice to have a story.  One solution for the developer would be to write a AppKit view in Swift that uses our SwiftUI button via `NSHostingView`.

Reply:
Tracking this with [a new issue](https://github.com/google/GoogleSignIn-iOS/issues/117).